lgc/builder/YCbCrConvertHandler.h

```diff
@@ -0,0 +1,272 @@
+ *  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ *  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ *  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ *  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ *  SOFTWARE.
+ *
+ **********************************************************************************************************************/
+/**
+ ***********************************************************************************************************************
+ * @file  YCbCrConvertHandler.h
+ * @brief LLPC header file: contains the definition of LLPC utility class YCbCrConvertHandler.
+ ***********************************************************************************************************************
+ */
+
+#pragma once
+
+#include "BuilderImpl.h"
+#include "YCbCrAddressHandler.h"
+#include "vkgcDefs.h"
+
+using Vkgc::SamplerYCbCrConversionMetaData;
+
+namespace lgc {
+
+// Represents the type of sampler filter.
+enum class SamplerFilter {
+  Nearest = 0,
+  Linear,
+};
+
+// Represents the YCbCr conversion model.
+enum class SamplerYCbCrModelConversion : uint32_t {
+  RgbIdentity = 0,
+  YCbCrIdentity,
+  YCbCr709,
+  YCbCr601,
+  YCbCr2020,
+};
+
+// Represents whether color channels are encoded using the full range of numerical
+// values or whether values are reserved for headroom and foot room.
+enum class SamplerYCbCrRange {
+  ItuFull = 0,
+  ItuNarrow,
+};
+
+// Represents the location of downsampled chroma channel samples relative to the luma samples.
+enum class ChromaLocation {
+  CositedEven = 0,
+  Midpoint,
+};
+
+// Represents the component values placed in each component of the output vector.
+class ComponentSwizzle {
+public:
+  enum Channel : uint32_t {
+    Zero = 0,
+    One = 1,
+    R = 4,
+    G = 5,
+    B = 6,
+    A = 7,
+  };
+
+  ComponentSwizzle() = default;
+  ComponentSwizzle(const Channel swizzle) : m_value(swizzle), m_channel(swizzle - Channel::R){};
+  ComponentSwizzle(const uint32_t swizzle) {
+    assert((swizzle >= Channel::Zero) && (swizzle <= Channel::A));
+    m_value = static_cast<Channel>(swizzle);
+
+    m_channel = (swizzle >= Channel::R) ? (m_value - Channel::R) : (m_value + Channel::R);
+  }
+
+  operator Channel() const { return m_value; }
+
+  ComponentSwizzle &operator=(uint32_t op) {
+    assert((op >= Channel::Zero) && (op <= Channel::A));
+    m_value = static_cast<Channel>(op);
+    m_channel = (op >= Channel::R) ? (m_value - Channel::R) : (m_value + Channel::R);
+    return *this;
+  }
+  constexpr bool operator!=(ComponentSwizzle::Channel op) const { return m_value != op; }
+  constexpr bool operator==(ComponentSwizzle::Channel op) const { return m_value == op; }
+
+  const uint32_t getChannel() const { return m_channel; }
```

Comment:
My clang errors on this, saying the first const has no effect [-Werror,-Wignored-qualifiers]. Please remove it.

Also, as Rex said, use unsigned rather than uint32_t.